code/modules/projectiles/guns/mods/mods.dm

```diff
@@ -571,13 +571,15 @@
 	..()
 	var/datum/component/item_upgrade/I = AddComponent(/datum/component/item_upgrade)
 	I.weapon_upgrades = list(
+		UPGRADE_BULK = 2,
 		GUN_UPGRADE_DAMAGE_BRUTE = 10,
 		GUN_UPGRADE_DAMAGE_BURN = 10,
 		GUN_UPGRADE_DAMAGE_TOX = 5,
 		GUN_UPGRADE_DAMAGE_OXY = 5,
 		GUN_UPGRADE_DAMAGE_CLONE = 5,
 		GUN_UPGRADE_DAMAGE_HALLOSS = 5,
-		UPGRADE_BULK = 1
+		GUN_UPGRADE_FIRE_DELAY_MULT = 1.2,
+		GUN_UPGRADE_CHARGECOST = 1.5
```

Comment:
what if you put it on a bullet gun? Wouldnt that error?

Reply:
Eh.. No? NanoAI consumes more power with used on an energy weapon. It should ignore it.

I’ll test it to make sure.

Reply:
Good point!